tools/autograd/templates/python_variable_methods.cpp

```diff
@@ -433,7 +433,7 @@ static PyObject * THPVariable_numpy(PyObject* self, PyObject* arg)
         "Can't call numpy() on Variable that requires grad. "
         "Use var.detach().numpy() instead.");
   }
-  return torch::utils::tensor_to_numpy(self_.tensor_data());
+  return torch::utils::tensor_to_numpy(self_.variable_data());
```

Comment:
instead of first checking whether the variable requires_grad and then calling `variable_data` -- doesn't it make more sense to move the check into `tensor_to_numpy` and then just pass the variable directly?

Reply:
Addressed in https://github.com/pytorch/pytorch/pull/22214.